### ISSUE TITLE
Fix missing AI Summary

### ISSUE THREAD

**Issue by nickoferrall** [ParabolInc/parabol#8200]
@tianrunhe reported in a recent meeting there were no AI Summaries (see [Slack message](https://parabol.slack.com/archives/C042E8PGPD4/p1683822527222089)).

In Sentry, we're getting some 429 errors from Open AI: https://parabol.sentry.io/issues/3912924421/events/?project=107196&referrer=issue-stream&statsPeriod=14d&stream_index=0

Here's a link to OpenAI's rate limits: https://platform.openai.com/account/rate-limits

It looks like there are two different types of rate limits:
1. Rate per minute (RPM)
2. Token per minute (TPM)

I think we have a few options:

- Apply for an increase in our limits
- Change models from `text-davinci-003` to `gpt-3.5-turbo`. This would increase the RPM threshold from 3,000 to 3,500, but the TPM limit would decrease from 250,000 to 90,000.
- See if there's a way to refactor the code to send fewer requests
- Implement [exponential backoff](https://platform.openai.com/docs/guides/rate-limits/retrying-with-exponential-backoff). I think this approach would be tricky for us as we want the AI Summaries soon after we've sent the request

I've emailed OpenAI requesting an increase in our limits.

While waiting for a response, I think we should improve our error handling to help us find out whether the 429 error is due to the RPM or TPM. If it's RPM, we should change models to ChatGPT. If it's TPM, we shouldn't.

If the limit increase request is rejected, we should look into refactoring the code to send fewer requests.

### AC

- [ ] Improve OpenAI error handling to see what type of 429 error we're getting

**Comment by tianrunhe** [comment]
My best guess is that we are suffering from the TPM issue. According to OpenAI's [doc](https://platform.openai.com/docs/models/gpt-3-5), `text-davinci-003` should also be able to handle 4096 tokens. However if I change our config from 256 to 4096, I get consistent 400 errors from OpenAI. I experienced the same issue when I tried to use GPT-3.5.

**Comment by tianrunhe** [comment]
I've found [this super helpful answer](https://community.openai.com/t/error-retrieving-completions-400-bad-request/34004/7) on the 400 errors. So basically the `max_tokens` we set when calling the API is to control the length of the completion/response. The max token limit (4,096) from the model = # of tokens from our input + # of tokens from the generated response (aka `max_tokens` we set). So no wonder if I set `max_tokens = 4096` I could consistently get 400 error because that essentially leaves no space for the inputs. This could also explain the error on the `master` branch: we set `max_tokens` to 256 but if the input size is huge (potentially for a large retro or simply just a huge topic with many reflections), it could very well exceed the 4,096 limit on the model level.

**Comment by nickoferrall** [comment]
> we might need a separate issue/approach to reduce input size

Yeah, I think that would be a good idea! 4,096 tokens are ~8,000 words or 4-5 pages of a book, so I imagine it's an edge case but still worth implementing 

> we could keep the 256 max_tokens as-is when calling OpenAI

I actually think decreasing the `max_tokens` limit could help us, and it's low handing fruit. I think that the `max_tokens` amount is added to the API per-minute limit, but as we only need ~40 tokens to generate 1-2 sentences, we're probably wasting tokens at the moment. We don't want to set it too low or the completion could be cut off, but perhaps a new limit of 80 could be a safe starting point?

**Comment by tianrunhe** [comment]
> We don't want to set it too low or the completion could be cut off, but perhaps a new limit of 80 could be a safe starting point?

I could imagine the whole meeting summary needs a bit more? But yeah agreed 80 is a good starting point for now

**Comment by nickoferrall** [comment]
> I could imagine the whole meeting summary needs a bit more?

The `max_tokens` refers only to the completion, i.e. the OpenAI response, so it doesn't matter how long the input is. For the whole summary, we're still requesting 1-2 sentences, so it should need only ~30 tokens. 



Also, pasting [my comment](https://github.com/ParabolInc/parabol/pull/8201#issuecomment-1561352349) here so we have all the research in one place:

It's still unclear whether we're hitting a token per minute limit (TPM) or a rate per minute (RPM) limit. Hopefully, we can find this out by:

- Looking at the frequency of usage of the API. If we're sending more than 3,000 requests per minute (the davinci limit [here](https://platform.openai.com/account/rate-limits)), we know it's a RPM limit. We should be able to see this in OpenAI's usage page ([here](https://platform.openai.com/account/usage)), but it's broken at the moment, saying "temporarily unavailable".
- Looking into our own meeting data to see how many summaries were created during a minute where we have a Sentry error.
- Changing max_tokens https://github.com/ParabolInc/parabol/pull/8267. If it results in significantly less 429 errors, we know it was a TPM problem.

I sent a request to OpenAI to increase our limits last week. They say they usually take 7-10 business days to respond, so we should hear from them this week or next.

**Comment by nickoferrall** [comment]
I haven't heard back from OpenAI. 

They have a [Usage Page](https://platform.openai.com/account/usage) that I mentioned in my previous comment, but that still says it's temporarily unavailable. 

Others are experiencing the same issue: https://community.openai.com/t/daily-usage-breakdown-does-not-load/176779

It did work for me once fairly recently. I'll keep revisiting for the next few days and see if it works again. We should be able to see which limit we're hitting by looking at these usage stats.

**Comment by nickoferrall** [comment]
Errors have largely stopped happening since July 7th so I'll close this issue: https://parabol.sentry.io/issues/3912924421/events/9dc3b00c301141c6a62edc4565534112/?project=107196

Rate limits haven't changed: https://platform.openai.com/account/rate-limits

I'm not sure why the errors have stopped happening. I previously saw a community note saying that the limits have been buggy for them, so perhaps this was an error on OpenAI's end as our usage hasn't changed.
